metrics/thresholds.go

```diff
@@ -188,6 +188,11 @@ func (ts *Thresholds) Run(sink Sink, duration time.Duration) (bool, error) {
 		ts.sinked["min"] = sinkImpl.Min
 		ts.sinked["max"] = sinkImpl.Max
 		ts.sinked["avg"] = sinkImpl.Avg
+
+		// Compute the median value of the trend sink
+		// We do this just in time, as it's needed, because it
+		// requires sorting the values, which computanionally expensive.
+		sinkImpl.Calc()
 		ts.sinked["med"] = sinkImpl.Med
```

Comment:
Looking at this my immediate reacto is to: 
1. just use `p(50)` here
2. drop `Med` field
3. drop `Calc` from everywhere
4. (optional) prioritize making `med` (and maybe all the rest of the threshold exposed fields) just in time calculated - so instead of you having a structure we always populate with some different fields. Have the threshold told us what it needs and then return that specific field, one at a time. 

WDYT?

Reply:
I totally agree, we reached the same conclusion with @imiric. We opened a dedicated issue to address refactoring the issues around `Calc` #2760.

From your perspective would you rather these changes be addressed in this specific PR, or would you be okay with merging this as is, and addressing the refactoring in a separate PR? (I'm fine with both)

Reply:
I would find 1-3 to be better done in this PR then in a following one.

4. is better of in a separate PR ... likely one with more tests and maybe even a benchmark

Reply:
Very good, let's do that 🤝

Reply:
Hey @oleiade, Sorry, but I see no changes :cry: probably push is missing

Reply:
Sorry @olegbespalov I forgot I had made the changes in a follow-up branch. My bad 🤦🏻